src/main/scala/edu/cmu/cs/obsidian/Main.scala

```diff
@@ -52,6 +54,7 @@ object Main {
         var printTokens = false
         var printAST = false
         var buildClient = false
+        var yul = false
```

Comment:
I'd rather have the target be an enum rather than a flag. It makes the API clearer — otherwise it's not obvious whether passing "true" results in Fabric or Yul output.